kernel/syscall.c

```diff
@@ -196,17 +196,55 @@ static bool _trace_syscall(long n)
     // Check if syscall tracing is enabled.
     if (__myst_kernel_args.strace_config.trace_syscalls)
     {
+        // -1 represents strace syscall filter is not set
+        short filter_syscall = -1;
+
         if (__myst_kernel_args.strace_config.filter)
         {
-            // If filtering is enabled, trace only this syscall has been
+            // If filtering is enabled, trace only this syscall if it has been
             // specified in the filter.
-            return __myst_kernel_args.strace_config.trace[n];
+            filter_syscall = __myst_kernel_args.strace_config.trace[n];
+
+            if (filter_syscall == 0)
+                return false;
         }
-        else
+        if (__myst_kernel_args.strace_config.tid_filter_num > 0)
         {
-            // Trace all syscalls.
-            return true;
+            // Check if thread filtering is enabled & follow filter
+            myst_thread_t* self = myst_thread_self();
+            pid_t current_tid = self->tid;
```

Comment:
Not should if its a good idea to `(long)self->tid` here, we don't have to cast to long in every loop. Sounds like over optimization

Reply:
Changed the data structure so we don't need long's at all